### PR TITLE
API-1328 - Using a custom formatter to scrub PII from all log records.

### PR DESCRIPTION
**Pull request by xlorepdarkhelm**
*A note to PR reviewers: it may be helpful to review our [code review documentation](https://github.com/GSA/notifications-api/blob/main/docs/all.md#code-reviews) to know what to keep in mind while reviewing pull requests.*

## Description

The `scrub()` method was restructured to be a function, and then applied to a new custom log formatter.

See: https://github.com/GSA/notifications-api/issues/1328

## Security Considerations

Scrubbing PII from log entries is very good for security purposes.

### CODE REVIEW

**Review by terrazoon — APPROVED** [review]

File: notifications_utils/logging.py [hunk 1]
```diff
@@ -18,6 +19,39 @@
...
+        masked_email = "XXXXX@XXXXXXX"
+        msg = msg.replace(email, masked_email)
+    return msg
+
+
+class PIIFilter(logging.Filter):
+    @override
+    def filter(self, record: logging.LogRecord) -> logging.LogRecord:
+        record.msg = _scrub(record.msg)
+        return record
```
Comment (ccostino) [inline notifications_utils/logging.py:46]: Is this in anticipation of being used elsewhere?

**Review by ccostino — COMMENTED** [review]
Awesome, thanks @xlorepdarkhelm!  Just a couple of quick questions from me!  One you'll see in the review, and then a general one about whether we ought to use [`re.compile`](https://docs.python.org/3/library/re.html#re.compile) for these regexes in the `_scrub` method.

Would that be any help or make things more complicated?

**Comment by xlorepdarkhelm** [comment]
> Awesome, thanks @xlorepdarkhelm! Just a couple of quick questions from me! One you'll see in the review, and then a general one about whether we ought to use [`re.compile`](https://docs.python.org/3/library/re.html#re.compile) for these regexes in the `_scrub` method.
> 
> Would that be any help or make things more complicated?

I answered the one question. Since you mentioned re.compile() in this, but didn't actually comment on it - yes, it makes perfect sense to use it, I just moved the code without looking at that. I adjusted the code now to use re.compile, which will save a lot of CPU cycles for us.

File: notifications_utils/logging.py (hunk 1 above)
Comment (ccostino) [reply notifications_utils/logging.py:48]: Ah okay, thanks!

File: notifications_utils/logging.py [hunk 2]
```diff
@@ -17,6 +18,41 @@
...
+    if not isinstance(msg, str):
+        return msg
+    phones = _phone_regex.findall(msg)
+
+    phones = [phone.replace("-", "").replace(" ", "") for phone in phones]
+    for phone in phones:
+        msg = msg.replace(phone, "1XXXXXXXXXX")
+
+    emails = re.findall(
+        r"[\w\.-]+@[\w\.-]+", msg
```
Comment (ccostino) [inline notifications_utils/logging.py:35]: D'oh, sorry, meant to include this regex along with the phone number for `re.compile`!

**Review by ccostino — COMMENTED** [review]
Thanks, again, @xlorepdarkhelm!  Sorry, I meant to include the email regex with the phone number for the `re.compile` question!

**Review by ccostino — APPROVED** [review]
Thanks, @xlorepdarkhelm!

**Pull request merged**